src/Process/Deploy/InstallUpdate/ConfigUpdate/SearchEngine.php

```diff
@@ -56,57 +47,69 @@ class SearchEngine implements ProcessInterface
      */
     private $searchEngineConfig;
 
+    /**
+     * @var ElasticSuite
+     */
+    private $elasticSuite;
+
     /**
      * @param LoggerInterface $logger
      * @param EnvWriter $envWriter
-     * @param EnvReader $envReader
      * @param SharedWriter $sharedWriter
-     * @param SharedReader $sharedReader
      * @param MagentoVersion $version
      * @param SearchEngineConfig $searchEngineConfig
+     * @param ElasticSuite $elasticSuite
      */
     public function __construct(
         LoggerInterface $logger,
         EnvWriter $envWriter,
-        EnvReader $envReader,
         SharedWriter $sharedWriter,
-        SharedReader $sharedReader,
         MagentoVersion $version,
-        SearchEngineConfig $searchEngineConfig
+        SearchEngineConfig $searchEngineConfig,
+        ElasticSuite $elasticSuite
     ) {
         $this->logger = $logger;
         $this->envWriter = $envWriter;
-        $this->envReader = $envReader;
         $this->sharedWriter = $sharedWriter;
-        $this->sharedReader = $sharedReader;
         $this->magentoVersion = $version;
         $this->searchEngineConfig = $searchEngineConfig;
+        $this->elasticSuite = $elasticSuite;
     }
 
     /**
      * Executes the process.
      *
      * @return void
+     * @throws ProcessException
      */
     public function execute()
     {
-        $this->logger->info('Updating search engine configuration.');
-
         $searchConfig = $this->searchEngineConfig->get();
 
+        $this->logger->info('Updating search engine configuration.');
         $this->logger->info('Set search engine to: ' . $searchConfig['engine']);
```

Comment:
This log will provide wrong information when ElasticSuite is installed

Reply:
Why?

Having this config
```
stage:
  global:
    SCD_ON_DEMAND: true
  deploy:
    ELASTIC_SUITE_CONFIGURATION:
      _merge: true
      engine:
        indices_settings:
          number_of_shards: 3
          number_of_replicas: 3
    SEARCH_CONFIGURATION:
      _merge: true
      engine: elasticsuite
```

Produces correct log

```
[2019-04-09 16:01:46] INFO: Set search engine to: elasticsuite  
```

Reply:
Try case when you have nothing in SEARCH_CONFIGURATION.
BTW, the value from this config is ignored as I see, so if you have nothing or engine was set as ES, you still set elasticsuite configs if module is installed.

Reply:
Covered by validator